### ISSUE TITLE
Unable to use hook created field in SQL WHERE clause

### ISSUE THREAD

**Issue by AntoineDcode** [orientechnologies/orientdb#5239]
Hi,

If a field of type "Long" is created inside a server side hook, field is successfully created but Select query return nothing.

**Edit**: Test case is working but not inside orient-studio 

Hook to deploy server side:
```java
public class IdGeneratorHook extends ODocumentHookAbstract implements ORecordHook {

    private IdGenerator generator = new LocalIdGeneratorImpl();

    public IdGeneratorHook() {
        setExcludeClasses("Log"); //if comment out this one line or leave off the constructor entirely then OrientDB fails on every command
    }

    @Override
    public DISTRIBUTED_EXECUTION_MODE getDistributedExecutionMode() {
        return DISTRIBUTED_EXECUTION_MODE.SOURCE_NODE;
    }

    @Override
    public RESULT onRecordBeforeCreate(ODocument doc) {
        doc.field("id", this.generator.get());

        return RESULT.RECORD_CHANGED;
    }
}
```

Test case:
```java
    @Test
    public void testHook() throws Exception {

        ODatabaseDocumentTx db = this.getDatabase();
        OSchema schema = db.getMetadata().getSchema();

        if (schema.existsClass("TestHook")) {
            schema.dropClass("TestHook");
        }
        OClass classA = schema.createClass("TestHook");

        classA.createProperty("id", OType.LONG);
        classA.createProperty("uuid", OType.STRING);
        classA.createProperty("id2", OType.LONG);
 
        // seems to work if registered on client
        //db.registerHook(new IdGeneratorHook()); 
        ODocument doc1 = new ODocument(classA);
        doc1.field("id2", 1);
        doc1.save();
        doc1.reload("*:1", true);

        System.out.println(doc1);

        Map<String, Object> params = new HashMap<>();
        params.put("id", doc1.field("id"));
        params.put("id2", doc1.field("id2"));

        List<ODocument> selectDoc = db.command(new OCommandSQL("SELECT * from TestHook")).execute();
        assertTrue(!selectDoc.isEmpty());
        assertEquals(params.get("id"), selectDoc.get(0).field("id"));
        assertEquals(params.get("id2"), selectDoc.get(0).field("id2"));

        List<ODocument> selectDocId2 = db.command(new OCommandSQL("SELECT * from TestHook where id2 = :id2")).execute(params);
        assertTrue(!selectDocId2.isEmpty());
        assertEquals(params.get("id2"), selectDocId2.get(0).field("id2"));

        List<ODocument> selectDocId = db.command(new OCommandSQL("SELECT * from TestHook where id = :id")).execute(params);
        assertTrue(!selectDocId.isEmpty());
    }
```

Maybe I'm missing something ?
Test has been done on OrientDb 2.1.3

Regards,

**Comment by tglman** [comment]
hi @AntoineDcode,

I cannot reproduce this, probably has been fixed long time ago, did you try the last 2.1 hotfix?

Regards

**Comment by tglman** [comment]
i've written tests on hooks on 2.2 as well, everything is working closing.
